### ISSUE TITLE
Try to prevent AppBase::quit() from being overridden by user app

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#1087]
This method is abstract virtual and both implemented by platform-specific sublasses and meant to be called by user. At minimal, we should move it out of the section of the header where methods live that should be overridden.

We can also make `quit()` non-virtual and call a more verbosely named abstract virtual method that is implemented by the platform subclasses (I've suggested `performQuit()` though that isn't very popular yet).

A different solution would be to use the [final specifier](http://en.cppreference.com/w/cpp/language/final), although this would mean we'd have to remember to use it in all existing and future AppBase subclasses to get a similar effect. Also the final specifier is not directly user facing as it is used in something like AppMsw.h, etc, so users won't find out that they weren't supposed to override that method until after they've tried and hit a compile error.

**Comment by num3ric** [comment]
Moving the definition + using final seems like a good approach (though we should make sure that the error messages are sufficiently explicit).

**Comment by paulhoux** [comment]
I agree with Eric, and thanks for coming up with these solutions, Rich.  The ```final``` keyword was designed for this situation.
